mesonbuild/backend/backends.py

```diff
@@ -1195,25 +1195,32 @@ def determine_windows_extra_paths(
         links to and return them so they can be used in unit
         tests.
         """
-        result: T.Set[str] = set()
         prospectives: T.Set[build.BuildTargetTypes] = set()
+        internal_deps: T.Set[str] = set()
+        external_deps: T.Set[str] = set()
+
         if isinstance(target, build.BuildTarget):
             prospectives.update(target.get_all_link_deps())
-            # External deps
-            result.update(self.extract_dll_paths(target))
 
         for bdep in extra_bdeps:
             prospectives.add(bdep)
             if isinstance(bdep, build.BuildTarget):
                 prospectives.update(bdep.get_all_link_deps())
+
         # Internal deps
         for ld in prospectives:
             dirseg = os.path.join(self.environment.get_build_dir(), self.get_target_dir(ld))
-            result.add(dirseg)
+            internal_deps.add(dirseg)
+
+        # External deps
+        if isinstance(target, build.BuildTarget):
```

Comment:
The next `if` tests the same condition. You should combine them and perform the test for_machine inside.